### PR TITLE
Bugfix/launcher default profile

### PR DESCRIPTION
**Pull request by MikeNeilson**
## Problem Description

The launcher wasn't opening the Setup properties editor correctly in the case of no, or the default, profile. Additionally discovered that if you launch the launcher with a profile it was only partially used as the default profile.

## Solution

Altered the behavior of launcher, cmdline args, and DecodesSetupFrame to take a profile instead of guessing what it should be.
Additionally (partially) centralized the logic for loading the DecodesSettings information from the profiles.

LauncherFrame and DecodesSetupFrame now require a valid Profile as an input parameter. The LauncherProfile stores the commandline profiled Profile as the "default" for the purposes for determining how to open the given applications.

## how you tested the change

Manually.
1. Open launcher (no profile set) and click Setup, default (user or decodes.properties) profile opens correctly
2. Open launcher (with profile set) and click Setup, selected profile opens correctly
3. Open launcher (with profile set), select (default) in the combo, selected profile opens correctly.

## Where the following done:

- [ ] Tests. Check all that apply:
   - [ ] Unit tests created or modified that run during ant test.
   - [ ] Integration tests created or modified that run during integration testing
         (Formerly called regression tests.)
   - [ ] Test procedure descriptions for manual testing
- [ ] Was relevant documentation updated?
- [ ] Were relevant config element (e.g. XML data) updated as appropriate

If you aren't sure leave unchecked and we will help guide you to want needs changing where.

### CODE REVIEW

File: src/main/java/decodes/launcher/DecodesPropsPanel.java [hunk 1]
```diff
@@ -217,44 +169,46 @@ public void loadFromSettings(DecodesSettings settings)
...
+            typ == DecodesSettings.DB_NWIS ? 2 :
+            typ == DecodesSettings.DB_CWMS ? 3 :
+            typ == DecodesSettings.DB_OPENTSDB ? 4 :
             typ == DecodesSettings.DB_HDB ? 5 : 0);
         editDbLocationField.setText(settings.editDatabaseLocation);
     }
 
-    public void saveToSettings(DecodesSettings settings)
+    public DecodesSettings saveToSettings()
     {
```
Comment (ktarbet) [inline src/main/java/decodes/launcher/DecodesPropsPanel.java:192]:
```suggestion
    DecodesSettings saveToSettings()
```

File: src/main/java/decodes/launcher/LauncherFrame.java [hunk 2]
```diff
@@ -2166,13 +2052,13 @@ protected void startProfileManager()
      */
     public Profile getSelectedProfile()
     {
-        return profileCombo == null ? null : (Profile)profileCombo.getSelectedItem();
+        return profileCombo == null ? launchProfile : (Profile)profileCombo.getSelectedItem();
     }
 
     private boolean trySendToProfileLauncher(String cmd)
     {
         profile = getSelectedProfile();
```
Comment (ktarbet) [inline src/main/java/decodes/launcher/LauncherFrame.java:2060]: profile here is null with a fresh XML database install.

The following error can be found with fresh XML install after clicking : Setup, or 'Decodes Database Editor' from the start_launcher main screen (without using profiles)

```text
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
        at decodes.launcher.LauncherFrame.trySendToProfileLauncher(LauncherFrame.java:2061)
        at decodes.launcher.LauncherFrame.setupPressed(LauncherFrame.java:962)
        at decodes.launcher.LauncherFrame$26.actionPerformed(LauncherFrame.java:677)
```

**Review by ktarbet — CHANGES_REQUESTED** [review]

File: src/main/java/decodes/launcher/LauncherFrame.java (hunk 2 above)
Comment (MikeNeilson) [reply src/main/java/decodes/launcher/LauncherFrame.java:2060]: Yep, the else clause in the profileCombo handling left the combo empty. I changed it to add the "launchProfile" item and set it as selected. Seems to behave now.

**Review by ktarbet — APPROVED** [review]

**Comment by MikeNeilson** [comment]
Given https://github.com/opendcs/opendcs/issues/476#issuecomment-1949114933 I'm going to wait to merge this in until I fix that particular behavior as well.

Partially a similar fix, needs to just use the profile as-provided, but also check that DbAuth *is* UserAuthFile as the others aren't supposed to be writable from the GUI. Might be worth adding a function to the AuthSourceProvider interface for checking writable and writing.

**Comment by MikeNeilson** [comment]
@agilmore2 

So I think the right approach to #476 is to just make a new izpack panel and prompt the user during install. Going through the docs that's going to involve creating a custom izpack panel. I don't think that's a huge deal but I don't think it makes sense
to do that in izpack 4.

I don't suggest inplace updates, but always using a fresh directory, so I feel like migrating to izpack5 is the correct to do now. 
But like a separate PR, this one is already a bit big and is normalizing at least some behavior between OSes.

**Comment by MikeNeilson** [comment]
I'm going to do the installer work as part of another PR, this at least improves operations once things have been configured.

**Pull request merged**